packages/flow-core/src/components/f-document-viewer/f-document-viewer.ts

```diff
@@ -0,0 +1,542 @@
+		function findDepth(obj: FDocViewerContent, depth: number) {
+			for (const key in obj) {
+				if (
+					typeof obj[key] === "object" &&
+					obj[key] !== null &&
+					(obj[key] as FDocumentStatement).data
+				) {
+					maxDepth = Math.max(maxDepth, depth);
+					findDepth((obj[key] as FDocumentStatement).data ?? {}, depth + 1);
+				}
+			}
+		}
+		findDepth(obj, 1);
+		return maxDepth;
+	}
+
+	/**
+	 *
+	 * @param key string for key
+	 * @param value string for value inside key could be an object or a string
+	 * @param innerHtml result in html to be rendered
+	 * @returns accordion panel template
+	 */
+	getAccordionPanel(key: string, value: FDocumentStatement, innerHtml: HTMLTemplateResult) {
+		return html` <f-accordion
+			icon-placement="left"
+			icon-size="x-small"
+			id=${this.refName(key)}
+			class="doc-preview-accordion"
+			?open=${value.open}
+			@toggle=${() => {
+				value.open = !value.open;
+			}}
+		>
+			${value.template
+				? html`<f-div height="100%">${value.template(this.searchValue)}</f-div>`
+				: html` <f-div gap="medium">
+						<f-div width="hug-content">
+							<f-text
+								variant="para"
+								size="small"
+								.weight=${this.getTextWeight(value.type)}
+								.state=${this.getState(value.type)}
+								id="doc-text"
+								.highlight=${this.searchValue}
+							>
+								${key}</f-text
+							></f-div
+						>
+						<f-div>
+							<f-text
+								variant="para"
+								size="small"
+								.weight=${this.getTextWeight(value.type)}
+								.state=${this.getState(value.type)}
+								id="doc-text"
+								.highlight=${this.searchValue}
+							>
+								${value.title}</f-text
+							></f-div
+						>
+				  </f-div>`}
+
+			<f-div slot="body" direction="column" width="100%">${innerHtml}</f-div>
+		</f-accordion>`;
+	}
+
+	/**
+	 *
+	 * @param key string for key
+	 * @param value string for value inside key
+	 * @returns html template
+	 */
+	getTextPanel(key: string, value: FDocumentStatement) {
+		return html` <f-div class="custom-body-padding" height="hug-content">
+			${value.template
+				? html`${value.template(this.searchValue)}`
+				: html` <f-div gap="medium" height="hug-content" gap="medium">
+						<f-div width="hug-content"
+							><f-text
+								variant="para"
+								size="small"
+								.weight=${this.getTextWeight(value.type)}
+								.state=${this.getState(value.type)}
+								id="doc-text"
+								.highlight=${this.searchValue}
+								>${key}</f-text
+							></f-div
+						>
+						<f-div
+							><f-text
+								variant="para"
+								size="small"
+								.state=${this.getState(value.type)}
+								id="doc-text"
+								.highlight=${this.searchValue}
+								>${value.title || value}</f-text
+							></f-div
+						>
+				  </f-div>`}
+		</f-div>`;
+	}
+
+	/**
+	 *
+	 * @param obj complete json object for doc creation
+	 * @returns template for the data given
+	 */
+	traverse(obj: FDocViewerContent) {
+		let htmlTemp = html``;
```

Comment:
We don't need `htmlTemp` here,

Check this approach
https://lit.dev/docs/templates/lists/#repeating-templates-with-map